src/main/java/org/opensearch/sdk/handlers/ExtensionDependencyResponseHandler.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.sdk.handlers;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.opensearch.common.io.stream.StreamInput;
+import org.opensearch.extensions.DiscoveryExtensionNode;
+import org.opensearch.extensions.ExtensionsManager;
+import org.opensearch.threadpool.ThreadPool;
+import org.opensearch.transport.TransportException;
+import org.opensearch.transport.TransportResponseHandler;
+
+/**
+ * This class handles the response from OpenSearch to a {@link ExtensionsRunner#sendExtensionDependencyRequest(TransportService)} call.
+ */
+public class ExtensionDependencyResponseHandler implements TransportResponseHandler<ExtensionDependencyResponse> {
+    private static final Logger logger = LogManager.getLogger(ExtensionDependencyResponseHandler.class);
+    private final CountDownLatch inProgressLatch;
+    private List<DiscoveryExtensionNode> extensions;
+
+    public ExtensionDependencyResponseHandler(){
+        this.inProgressLatch = new CountDownLatch(1);
```

Comment:
Can you replace the CountDownLatch with a CompletableFuture? See https://github.com/opensearch-project/OpenSearch/pull/5646 for how to do this

Reply:
sure

Reply:
I also just raised a PR here on the SDK that does the same thing for the other latches: https://github.com/opensearch-project/opensearch-sdk-java/pull/301